bpo-38981: Rename re.error to re.ReCompileError for better readability. (#17501)
python/cpython: Doc/library/re.rst

   Clear the regular expression cache.

> .. exception:: ReCompileError(msg, pattern=None, pos=None)

Reviewer: We need an explicit index reference for `error`.

Author: how would I add and explicit index reference ? `.. error:` on top ?

   Exception raised when a string passed to one of the functions here is not a
   valid regular expression (for example, it might contain unmatched parentheses)
   or when some other error occurs during compilation or matching.  It is never an

>    error if a string contains no match for a pattern.  The `ReCompileError` instance has

Reviewer 2: It appears that this line should be re-wrapped.

>    error if a string contains no match for a pattern.  The `ReCompileError` instance has

Reviewer: Use ``` :exc:`!...` ``` or double backticks.

   .. versionchanged:: 3.5
      Added additional attributes.

>    .. versionchanged:: 3.9

Reviewer: Use double backticks around new and old names.

>       ReCompileError was previsouly named 'error', the alias has been kept for

Reviewer 2: ```suggestion
      ReCompileError was originally named 'error'; the latter is kept as an alias for
```
